core/app/models/concerns/spree/active_storage_adapter/attachment.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+require 'mini_magick'
+
+module Spree
+  module ActiveStorageAdapter
+    # Decorares AtiveStorage attachment to add methods exptected by Solidus'
+    # Paperclip-oriented attachment support.
+    class Attachment
+      delegate_missing_to :@attachment
+
+      DEFAULT_SIZE = '100%'
+
+      def initialize(attachment, styles: {})
+        @attachment = attachment
+        @styles = styles
+      end
+
+      def exists?
+        attached?
+      end
+
+      def filename
+        blob.filename.to_s
+      end
+
+      def url(style = nil)
+        variant(style).url
+      end
+
+      def variant(style = nil)
+        size = style_to_size(style&.to_sym)
+        @attachment.variant(
+          resize: size,
+          strip: true,
+          'auto-orient': true,
+          colorspace: 'sRGB',
+        ).processed
+      end
+
+      def height
+        metadata[:height]
+      end
+
+      def width
+        metadata[:width]
+      end
+
+      def destroy
+        return false unless attached?
+
+        purge
+        true
+      rescue StandardError => e
```

Comment:
Are you 100% sure we want to rescue StandardError? This seems very dangerous as there's a ton of things that may go wrong it's probably not a good idea to swallow them silently.

Reply:
There are no explicit errors defined Rails side but `ActiveStorage::Service` implementations could raise their own errors. To be safe I'm removing the rescue altogether 👍